### ISSUE TITLE
fk_locationtype

### ISSUE THREAD

**Issue by tudorbarascu** [qwat/qwat-data-model#183]
On my way on migrating to the newest model, I noticed the pipe ``fk_locationtype`` lots of empty arrays ``{}`` and ``NULLS``.

Further investigating it seems empty arrays are always created when editing through QGIS **Value relation** widget. Checking or not the **Allow NULL value**  doesn't change anything.

Do you think that we should search for consistency here? (e.g. where no one completed anything database side should be either NULL or {})

I would have expected the widget to insert NULL values as everything is empty and no one touched it.

Thoughts?

**Comment by tudorbarascu** [comment]
in QGIS 3. custom UI doesn't work for fk_locationtype (arrays..) Issue https://issues.qgis.org/issues/20714

**Comment by ponceta** [comment]
We do not use `fk_locationtype` in Pully, If there's a bug with the value relation widget, it could be interesting to check with @dsavary, he's using much more than I do now.

**Comment by tudorbarascu** [comment]
@haubourg As I read somewhere on issues.qgis.org you're updating the qwat.qgs project to version 3, for the fk_locationtype in the .ui files you need to change the type from list widget to ``table widget``.
Any timeframe/deadline for updating as I would like to update the VM that holds qwat/qgep.
Thank you! We can close this issue if it doesn't serve others as it's solved from my part. More details here: https://issues.qgis.org/issues/20714

**Comment by haubourg** [comment]
@tudorbarascu In fact, I'm evaluating the work for a complete migration and move the forms from UI to drag and drop designer. 

@ponceta any will on your side to push forward that migration work ? 

>  you need to change the type from list widget to `table widget`.

If we go to a drag and drop form, I suppose we will workaround that issue right?

**Comment by tudorbarascu** [comment]
Yes.. that would surely work. Last time I tested the drag&drop forms were loading considerably slower than UI-s. I haven't tested in the last few months and things may have changed.. so..just FYI.

**Comment by haubourg** [comment]
@tudorbarascu ok, I'll focus on that performance part then, thanks!

**Comment by ponceta** [comment]
> @tudorbarascu In fact, I'm evaluating the work for a complete migration and move the forms from UI to drag and drop designer.

That would be great yes!

> @ponceta any will on your side to push forward that migration work ?

Like discussed, migration is in progress, some 3D stuff and IT ressources are still blocking us from getting full QGIS 3. But it's a matter of weeks.

> Yes.. that would surely work. Last time I tested the drag&drop forms were loading considerably slower than custom UI-s. I haven't tested in the last few months and things may have changed.. so..just FYI.

QGIS 3 is a lot faster than QGIS 2 on many relationnal widgets. We want to adress this issue since a long time with @dsavary, there are many points for improvement, we will need to have a strategy to solve these progressively.

Do we close or rename / reopen a(some) new issue(s) for all good points mentionned here above?

**Comment by tudorbarascu** [comment]
@ponceta 
> Do we close or rename / reopen a(some) new issue(s) for all good points mentionned here above?

Whatever looks more suiting to you, I don't have any preference.

>QGIS 3 is a lot faster than QGIS 2 on many relationnal widgets.

The tests and the speed difference were observed on QGIS 3.
